src/Parsers/ASTTableOverrides.cpp

```diff
@@ -0,0 +1,242 @@
+    String hl_none = settings.hilite ? hilite_none : "";
+
+    settings.ostr << hl_keyword << "TABLE OVERRIDE " << hl_none;
+    ASTIdentifier(table_name).formatImpl(settings, state, frame);
+    if (!columns && (!storage || storage->children.empty()))
+        return;
+    auto override_frame = frame;
+    ++override_frame.indent;
+    settings.ostr << nl_or_ws << '(' << nl_or_nothing;
+    String indent_str = settings.one_line ? "" : String(4 * override_frame.indent, ' ');
+    size_t override_elems = 0;
+    if (columns)
+    {
+        FormatStateStacked columns_frame = override_frame;
+        columns_frame.expression_list_always_start_on_new_line = true;
+        settings.ostr << indent_str << hl_keyword << "COLUMNS" << hl_none << nl_or_ws << indent_str << "(";
+        columns->formatImpl(settings, state, columns_frame);
+        settings.ostr << nl_or_nothing << indent_str << ")";
+        ++override_elems;
+    }
+    if (storage)
+    {
+        const auto & format_storage_elem = [&](IAST * elem, const String & elem_name)
+        {
+            if (elem)
+            {
+                settings.ostr << (override_elems++ ? nl_or_ws : "")
+                              << indent_str
+                              << hl_keyword << elem_name << hl_none << ' ';
+                elem->formatImpl(settings, state, override_frame);
+            }
+        };
+        format_storage_elem(storage->partition_by, "PARTITION BY");
+        format_storage_elem(storage->primary_key, "PRIMARY KEY");
+        format_storage_elem(storage->order_by, "ORDER BY");
+        format_storage_elem(storage->sample_by, "SAMPLE BY");
+        format_storage_elem(storage->ttl_table, "TTL");
+    }
+
+    settings.ostr << nl_or_nothing << ')';
+}
+
+void ASTTableOverride::applyToCreateTableQuery(ASTCreateQuery * create_query) const
+{
+    if (columns)
+    {
+        if (!create_query->columns_list)
+            create_query->set(create_query->columns_list, std::make_shared<ASTColumns>());
+        if (columns->columns)
+        {
+            for (const auto & override_column_ast : columns->columns->children)
+            {
+                auto * override_column = override_column_ast->as<ASTColumnDeclaration>();
+                if (!override_column)
+                    continue;
+                if (!create_query->columns_list->columns)
+                    create_query->columns_list->set(create_query->columns_list->columns, std::make_shared<ASTExpressionList>());
+                auto & dest_children = create_query->columns_list->columns->children;
+                auto exists = std::find_if(dest_children.begin(), dest_children.end(), [&](ASTPtr node) -> bool
+                {
+                    return node->as<ASTColumnDeclaration>()->name == override_column->name;
+                });
+                if (exists == dest_children.end())
+                    dest_children.emplace_back(override_column_ast);
+                else
```

Comment:
Will break replication if we add a column which exists in table override and does not exist in mysql? Haven't read how new columns are inserted in mysql replication though, but in postgresql replication it will break with current code, though with tiny change this can be fixed, but anyway why need to add such column at all?.

Reply:
The current MySQL replication will convert ALTER TABLE ADD COLUMN queries fine, but if you add a column in an override first, then add it in MySQL, replication will break with my PR. IMHO this is within "can be documented" for now, along with many other ways one can shoot ones foot. It's not ideal, but it's not straightforward to me what a better behavior would be.

There are good use cases for adding materialized columns, I tried to add semi-meaningful examples in the docs.

Reply:
Ok, I agree.

Reply:
> IMHO this is within "can be documented"

Then need to add it now in a noticable place as a warning in doc? I don't see in current doc.

Reply:
Added in ada16f6ebe3a9a514fd6c1ec5966b1392a17af6d